### PR TITLE
[Do Not Merge] Reorganized uppergamma docstring with usage at top.

### PR DESCRIPTION
**Pull request by moorepants**
#### Brief description of what is fixed or changed

Example of a docstring that makes the programmatic use at the top and moves the rendered latex into the notes section below.

This is an example, do not merge.

### CODE REVIEW

**Comment by moorepants** [comment]
Rendered:
Selection_546

**Comment by moorepants** [comment]
IPython help:
```
Init signature: uppergamma(a, z)
Docstring:     
Class that represents the unevaluated upper incomplete gamma function
[1]_. Evaluation of the upper gamma function Γ(s, z) requires two
arguments, where s is the exponent of the integrand and x is the lower
limit of the integral. See the notes below for mathematical definitions.

Examples
========

>>> from sympy import uppergamma, S
>>> from sympy.abc import s, x
>>> uppergamma(s, x)
uppergamma(s, x)
>>> uppergamma(3, x)
2*(x**2/2 + x + 1)*exp(-x)
>>> uppergamma(-S(1)/2, x)
-2*sqrt(pi)*erfc(sqrt(x)) + 2*exp(-x)/sqrt(x)
>>> uppergamma(-2, x)
expint(3, x)/x**2

See Also
========

gamma: Gamma function.
lowergamma: Lower incomplete gamma function.
polygamma: Polygamma function.
loggamma: Log Gamma function.
digamma: Digamma function.
trigamma: Trigamma function.
sympy.functions.special.beta_functions.beta: Euler Beta function.

Notes
=====

The upper incomplete gamma function can be defined as the meromorphic
continuation of

.. math::
    \Gamma(s, x) := \int_x^\infty t^{s-1} e^{-t} \mathrm{d}t =
      \Gamma(s) - \gamma(s, x)\textrm{.}

where `\gamma(s, x)` is the lower incomplete gamma function,
:class:`lowergamma`. This can be shown to be the same as

.. math::
    \Gamma(s, x) = \Gamma(s) -
      \frac{x^s}{s} {}_1F_1\left({s \atop s+1} \middle| -x\right)\textrm{,}

where :math:`{}_1F_1` is the (confluent) hypergeometric function.

The upper incomplete gamma function is also essentially equivalent to the
generalized exponential integral:
```

**Comment by moorepants** [comment]
Python `help(uppergamma)`:
```
Help on class uppergamma in module sympy.functions.special.gamma_functions:

class uppergamma(sympy.core.function.Function)
 |  Class that represents the unevaluated upper incomplete gamma function
 |  [1]_. Evaluation of the upper gamma function Γ(s, z) requires two
 |  arguments, where s is the exponent of the integrand and x is the lower
 |  limit of the integral. See the notes below for mathematical definitions.
 |  
 |  Examples
 |  ========
 |  
 |  >>> from sympy import uppergamma, S
 |  >>> from sympy.abc import s, x
 |  >>> uppergamma(s, x)
 |  uppergamma(s, x)
 |  >>> uppergamma(3, x)
 |  2*(x**2/2 + x + 1)*exp(-x)
 |  >>> uppergamma(-S(1)/2, x)
 |  -2*sqrt(pi)*erfc(sqrt(x)) + 2*exp(-x)/sqrt(x)
 |  >>> uppergamma(-2, x)
 |  expint(3, x)/x**2
 |  
 |  See Also
 |  ========
 |  
 |  gamma: Gamma function.
 |  lowergamma: Lower incomplete gamma function.
 |  polygamma: Polygamma function.
 |  loggamma: Log Gamma function.
 |  digamma: Digamma function.
 |  trigamma: Trigamma function.
 |  sympy.functions.special.beta_functions.beta: Euler Beta function.
 |  
 |  Notes
 |  =====
 |  
 |  The upper incomplete gamma function can be defined as the meromorphic
 |  continuation of
 |  
 |  .. math::
 |      \Gamma(s, x) := \int_x^\infty t^{s-1} e^{-t} \mathrm{d}t =
 |        \Gamma(s) - \gamma(s, x)\textrm{.}
 |  
 |  where `\gamma(s, x)` is the lower incomplete gamma function,
 |  :class:`lowergamma`. This can be shown to be the same as
 |  
 |  .. math::
 |      \Gamma(s, x) = \Gamma(s) -
 |        \frac{x^s}{s} {}_1F_1\left({s \atop s+1} \middle| -x\right)\textrm{,}
 |  
 |  where :math:`{}_1F_1` is the (confluent) hypergeometric function.
 |  
 |  The upper incomplete gamma function is also essentially equivalent to the
 |  generalized exponential integral:
```

**Comment by moorepants** [comment]
SymPy Function classes also offer a unique documentation challenge, because you really need to documention how `__eval()__` works. From this class docstring, you only discern this from the example and it is awkard to add a `Parameters` and `Returns` section to the class level docstring when it is really describing the `__eval__()` method.

**Comment by certik** [comment]
@moorepants thanks. I like this. @asmeurer, what do you think?

**Comment by asmeurer** [comment]
I like it better the way it was, with the function definition at the top. This also breaks some other docstring rules, like the first line should be a short sentence. 

> SymPy Function classes also offer a unique documentation challenge, because you really need to documention how __eval()__ works. From this class docstring, you only discern this from the example and it is awkard to add a Parameters and Returns section to the class level docstring when it is really describing the __eval__() method.

The fact that `eval` is used is an implementation detail that only matters if you are writing code on the class. For end users, `uppergamma` is just a mathematical SymPy function. Even the fact that it is a class is irrelevant to most.

File: sympy/functions/special/gamma_functions.py [hunk 1]
```diff
@@ -342,27 +342,10 @@ def _eval_rewrite_as_expint(self, s, x, **kwargs):
...
-
-    The upper incomplete gamma function is also essentially equivalent to the
-    generalized exponential integral:
-
-    .. math::
-        \operatorname{E}_{n}(x) = \int_{1}^{\infty}{\frac{e^{-xt}}{t^n} \, dt} = x^{n-1}\Gamma(1-n,x).
+    r"""Class that represents the unevaluated upper incomplete gamma function
+    [1]_. Evaluation of the upper gamma function Γ(s, z) requires two
+    arguments, where s is the exponent of the integrand and x is the lower
+    limit of the integral. See the notes below for mathematical definitions.
```
Comment (asmeurer) [inline sympy/functions/special/gamma_functions.py]: I'm not sure if this is how the parameters are usually named. 

In general, for mathematical functions, I don't think it's helpful to describe the parameters in plain English. It's much more useful to see the mathematical definition.

File: sympy/functions/special/gamma_functions.py [hunk 2]
```diff
@@ -389,16 +372,46 @@ class uppergamma(Function):
...
 
+    Notes
+    =====
+
+    The upper incomplete gamma function can be defined as the meromorphic
+    continuation of
+
+    .. math::
+        \Gamma(s, x) := \int_x^\infty t^{s-1} e^{-t} \mathrm{d}t =
+          \Gamma(s) - \gamma(s, x)\textrm{.}
```
Comment (asmeurer) [inline sympy/functions/special/gamma_functions.py]: Does `\textrm{.}` do anything?
